### ISSUE TITLE
detecting rectangle (is_rectangle)

### ISSUE THREAD

**Issue by yunhal** [tdmorello/napari-geojson#7]
Can I request a working function of is_rectangle? 
One of my GEOJSON data has 4 points, which is currently classified as polygon and it leads napari to crash.

**Comment by tdmorello** [comment]
Yes, absolutely. Can you send me the geojson file or a geojson file with the shape that causes it to crash?

**Comment by yunhal** [comment]
I really appreciate your quick response! 

I attached my geojson file here - please remove ".txt" ending before using it. Side note that this file seems to have some error in data (e.g., weirdness in bottom part of South Africa). However it should be enough for improving "is_rectangle" function. 

Thanks again! 


[world.geojson.txt](https://github.com/tdmorello/napari-geojson/files/11064570/world.geojson.txt)

**Comment by yunhal** [comment]
The incorrect shape in the bottom part of South Africa seems to be caused by napari-geojson.  Using the same source geojson file, I can get the correct map shapes in other library such as geopandas.  Can you also help on this?

**Comment by tdmorello** [comment]
Hi @yunhal , can you tell me which version of napari you are using?

**Comment by yunhal** [comment]
Hi Tim.  Here is my napari environment information:

napari: 0.4.16
Platform: macOS-12.6-x86_64-i386-64bit
System: MacOS 12.6
Python: 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:43:44) [Clang 13.0.1 ]
Qt: 5.15.4
PyQt5: 5.15.7
NumPy: 1.23.3
SciPy: 1.9.1
Dask: 2022.9.2
VisPy: 0.10.0
OpenGL:

GL version: 2.1 ATI-4.8.101
MAX_TEXTURE_SIZE: 16384
Screens:

screen 1: resolution 1920x1080, scale 2.0
Plugins:

brainglobe-napari-io: 0.1.5
brainreg-napari: 0.0.5rc0
brainreg-segment: 0.2.16
console: 0.0.6
napari-animation: 0.0.5
napari-dataset: 0.1.dev81+g454e834
napari-geojson: 0.1.3
napari-svg: 0.1.6
ndtiffs: 0.1.2
scikit-image: 0.4.16

**Comment by psobolewskiPhD** [comment]
I think the crashing here was due to the hole in the polygon in south africa due to napari triangulation. 
Rect vs polygon doesn't really matter. But of course detecting shape type would be nice!

**Comment by psobolewskiPhD** [comment]
Note: napari apparently doesn't validate that a rectangle is a rectangle:
https://github.com/napari/napari/issues/4289

And GeoJSON does not have a Rectangle concept, just polygon. But it's not hard to check if a polygon is a rectangle and then set the shape_type in napari. I'm not sure there is a real functional difference in napari between a polygon and a rectangle TBH.
